jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/GeneralParams.scala

```diff
@@ -171,7 +179,7 @@ private[spark] trait GeneralParams extends Params {
 
   final def getSeed: Long = $(seed)
 
-  setDefault(numRound -> 1, numWorkers -> 1, nthread -> 1,
+  setDefault(numRound -> 1, numWorkers -> 1, nthread -> 1, nGpus -> 1,
```

Comment:
this should be 0 at least

Reply:
I'd be happy to change it, but it's only used when `tree_method` is set to `gpu_hist`, in which case the user probably expects to grab a GPU. It's also the default on the C++ side (https://github.com/dmlc/xgboost/blob/master/src/tree/param.h#L200), so having this default seems less surprising and more consistent.